File: Functions/update_budget.c
#include <stdio.h>
float income, rent, utilities, groceries, transportation, expenses, savings, total;

float input(char type[], float var){
    printf("how much is your %s: \n", type);
    scanf("%f", &var);
    return var;
};

void percent(char type[],int amount){
   int per = amount/income*100;
   printf("your %s is %d%% of your income.\n", type, per);
}

int main(void){
    printf("This is going to calculate your budget for the month\n");
    income = input("income", income);
    rent = input("rent",rent);
    utilities = input("utilities",utilities);
    groceries = input("groceries", groceries);
   transportation = input("transportation", transportation);

    expenses = rent + utilities + groceries + transportation;
    savings = income *.2;
    total = income - expenses - savings;
    printf("you make $%.2f\n", income);
    printf("you save $%.2f\n", savings);
    printf("you have $%.2f\n", total);
    
    percent("rent", rent);
    percent("utilities", utilities);
    percent("groceries", groceries);
    percent("transportation", transportation);
    percent("savings", savings);
    percent("expenses", expenses);
    
    
 return 0;   
}

File: Functions/update_budget.py

def scanf(type):
    
    return input(f"how much is your {type}: ")

print("Hello, and welcome to your finacial calculator!\n")
income = float (scanf("income"))
utilities = float (scanf("utilities"))
rent= float (scanf("rent"))
groceries = float (scanf("groceries"))
transportation = float (scanf("transportation"))
expenses = rent + utilities + groceries + transportation
savings = income * 0.2
total = income - expenses - savings

def percent(type,amount):

    per = amount/income*100

    print(f"your {type} is {per}% income")


print("your income",income)
print("your expenses",expenses)
print("your savings",savings)
print("your total",total)
percent("rent",rent)
percent("utilities",utilities)
percent("groceries",groceries)
percent("transport",transportation)
percent("expenses",expenses)